#! /bin/sh /usr/share/dpatch/dpatch-run
## 351372_mipsel_ftbfs.dpatch by Paul Wise <pabs3@bonedaddy.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix #351372: mipsel FTBFS

@DPATCH@
diff -urNad gdal-1.3.1~/frmts/pcraster/libcsf/csftypes.h gdal-1.3.1/frmts/pcraster/libcsf/csftypes.h
--- gdal-1.3.1~/frmts/pcraster/libcsf/csftypes.h	2005-09-29 04:54:53.000000000 +0800
+++ gdal-1.3.1/frmts/pcraster/libcsf/csftypes.h	2006-02-08 00:10:55.000000000 +0800
@@ -91,19 +91,22 @@
 #endif
 
 /* endian mode
- * DEFINE WITH -D
- * first probe a few:
+ * DEFINE WITH -D or find here
  */
 #ifndef CPU_BIG_ENDIAN
 # ifndef CPU_LITTLE_ENDIAN
 
-/* GDAL CPL STYLE */
-#ifdef CPL_LSB
-# define CPU_LITTLE_ENDIAN
-#endif
-#ifdef CPL_MSB
-# define CPU_BIG_ENDIAN
-#endif
+#ifdef USE_IN_GDAL
+  /* GDAL CPL STYLE */
+# ifdef CPL_LSB
+#  define CPU_LITTLE_ENDIAN
+# endif
+# ifdef CPL_MSB
+#  define CPU_BIG_ENDIAN
+# endif
+
+#else
+ /* probe a few: */
 
 #ifdef _AIX
 /* IBM AIX defines this on RS/6000 */
@@ -115,10 +118,13 @@
 # define CPU_BIG_ENDIAN
 #endif
 
-#ifdef mips
-/* works on the SGI machines */
-# define CPU_BIG_ENDIAN
-#endif
+/*
+ * #ifdef mips
+ * worked once on the SGI machines
+ * but mips has both endian architectures
+ * # define CPU_BIG_ENDIAN
+ * #endif
+ */
 
 #ifdef __alpha
 /* DEC alpha defines this
@@ -145,6 +151,9 @@
 #  define CPU_BIG_ENDIAN
 #endif
 
+/* endif probing */
+# endif 
+
 /* endif no ENDIAN defined */
 # endif
 #endif
